aiinfra-cluster/modules/gke-cluster/main.tf

```diff
@@ -0,0 +1,284 @@
+  pod_security_policy_config {
+    enabled = true
+  }
+
+  # Enable shielded nodes to meet go/gke-cluster-pattern#req1.1.5
+  enable_shielded_nodes = true
+
+  cluster_autoscaling {
+    autoscaling_profile = "OPTIMIZE_UTILIZATION"
+    # The name of this attribute is very misleading, it controls node
+    # autoprovisioning (NAP), not autoscaling.
+    enabled = false
+  }
+
+  network_policy {
+    # Enabling NetworkPolicy for clusters with DatapathProvider=ADVANCED_DATAPATH
+    # is not allowed. Dataplane V2 will take care of network policy enforcement
+    # instead.
+    enabled = false
+    # GKE Dataplane V2 support. This must be set to PROVIDER_UNSPECIFIED in
+    # order to let the datapath_provider take effect.
+    # https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/issues/656#issuecomment-720398658
+    provider = "PROVIDER_UNSPECIFIED"
+  }
+
+  private_cluster_config {
+    enable_private_endpoint = true
+    enable_private_nodes    = true
+    master_ipv4_cidr_block  = local.master_cidr_block
+    master_global_access_config {
+      enabled = false
+    }
+  }
+
+  # Adding this block enables IP aliasing.
+  ip_allocation_policy {
+    cluster_ipv4_cidr_block  = local.pod_cidr_block == "" ? null : local.pod_cidr_block
+    services_ipv4_cidr_block = local.service_cidr_block == "" ? null : local.service_cidr_block
+  }
+
+  # This change will also enable the metadata server on nodes.
+  # go/gke-cluster-pattern#req4.1.1#req1.1.5 (parts of, vTPM is another section)
+  workload_identity_config {
+    workload_pool = "${var.project}.svc.id.goog"
+  }
+
+  authenticator_groups_config {
+    # Contact safer-gcp to get your group whitelisted for access.
+    # Beta feaure: don't depend on it for breakglass access.
+    security_group = "gke-security-groups@google.com"
+  }
+
+  # GKE Dataplane V2 support
+  # https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/container_cluster#datapath_provider
+  datapath_provider = var.enable_dataplane_v2 ? "ADVANCED_DATAPATH" : "DATAPATH_PROVIDER_UNSPECIFIED"
+
+  # regular release is required for all 1.24+ features.
+  release_channel {
+    channel = "REGULAR"
+  }
+
+  # Allow planned maintenance during this time period.
+  maintenance_policy {
+    recurring_window {
+      start_time = "2023-01-01T09:00:00Z"
+      end_time = "2025-01-01T17:00:00Z"
+      # On the 1st Tuesday of every 6 months.
+      recurrence = "FREQ=MONTHLY;INTERVAL=6;BYDAY=1TU"
+    }
+  }
+
+  addons_config {
+    # We optionally enable the Istio Add-on. Istio is required if there is
+    # any pod-to-pod communication. We keep it optional
+    # to simplify small deployments where there is no pod-to-pod communication
+    # (for now)
+    #
+    # go/gke-cluster-pattern#req5.2.1 for internal traffic, and satified
+    # only if var.enable_istio = "true".
+    istio_config {
+      # Awkward disable flag.
+      disabled = false
```

Comment:
Does this still mean "disabled"?

Reply:
No, this is enabled.

Reply:
The comment above makes me think that the intention is to disable it. Could you change the comment please?

Reply:
updated comment.